File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Island Perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        n, m = len(grid), len(grid[0])
        ans, neg = 0, 0

        for i in range(n):
            for j in range(m):
                if grid[i][j]:
                    ans += 1
                    if i < n-1 and grid[i+1][j]:
                        neg += 1
                    if j < m-1 and grid[i][j+1]:
                        neg += 1
        return ans*4 - neg*2

File: Online Judges/LeetCode/July LeetCoding Challenge/Week-1/Island Perimeter.rs
impl Solution {
    pub fn island_perimeter(grid: Vec<Vec<i32>>) -> i32 {
        let mut ans:i32 = 0;
        let mut neg:i32 = 0;
        let n  = grid.len() as usize;
        let m = grid[0].len() as usize;
        
        for i in 0..n{
            for j in 0..m{
                if grid[i][j] == 1 {
                    ans += 1;
                    if i < n-1 && grid[i+1][j] == 1 {
                        neg += 1;
                    }
                    if j<m-1 && grid[i][j+1] == 1 {
                        neg += 1;
                    }
                }
            }
        }
        return ans*4 - neg*2;
    }
}